src/storage/src/source/source_reader_pipeline.rs

```diff
@@ -343,14 +343,14 @@ where
                                 SourceMessageType::Finalized(message) | SourceMessageType::InProgress(message) => {
                                     let pid = message.partition.clone();
                                     let offset = message.offset;
-                                    // advance the _offset_ frontier if this the final message for that offset
+                                    // Advance the offset _frontier_ to past this message's offset.
+                                    // This will not affect the upper of the data shard(s) until
+                                    // the `reclock` operator processes ALL batches at this offset.
                                     //
-                                    // TODO(guswynn): when we remove this, we can simplify some of
+                                    // TODO(guswynn): We can simplify some of
```

Comment:
@aljoscha i forget what we suggested we do here, but i think our future plans subsume this